src/Shared/Distribt.Shared.EventSourcing/EventSourcingDependencyInjection.cs

```diff
@@ -13,7 +13,15 @@ public static void AddMongoEventSourcing(this IServiceCollection serviceCollecti
         //TODO: probably here it should be the addmongodb thingy
         serviceCollection.AddTransient(typeof(IAggregateRepository<>), typeof(AggregateRepository<>));
         serviceCollection.AddTransient<IEventStore, EventStore>();
-        serviceCollection.AddTransient<IEventStoreManager, MongoEventStoreManager>();
+        serviceCollection.AddTransient<IEventStoreManager>((IServiceProvider serviceProvider) =>
+        {
+            var mongoDbConnectionProvider = serviceProvider.GetService<IMongoDbConnectionProvider>()
+                                            ?? throw new ArgumentNullException(nameof(IMongoDbConnectionProvider));
```

Comment:
I hate exceptions, and in this case is not an exception 😂; 

you can do getrequiredservice instead of getservice and it will break if it is not there; 

this said, i think we can remove all of this getservices thingy, and think about the wrapper on the mongoclient as that will fix the issue of passing the mongourl to the `mongoEventStoreManager` too.